gdal/ogr/ogrsf_frmts/gpkg/ogrgeopackagetablelayer.cpp

```diff
@@ -806,6 +809,79 @@ OGRErr OGRGeoPackageTableLayer::ReadTableDefinition()
         }
     }
 
+#if !defined(__GNUC__) || defined(__clang__) || __GNUC__ >= 5
+
+    // Unique fields detection
+    char* pszTableDefinitionSQL = sqlite3_mprintf("select sql from sqlite_master where type='table' and name='%q'", m_pszTableName);
+    err = SQLQuery(poDb, pszTableDefinitionSQL, &oResultTable);
+    sqlite3_free(pszTableDefinitionSQL);
+
+    if ( err != OGRERR_NONE || oResultTable.nRowCount == 0 )
+    {
+        if( oResultTable.pszErrMsg != nullptr )
+            CPLError( CE_Failure, CPLE_AppDefined, "%s", oResultTable.pszErrMsg );
+        else
+            CPLError( CE_Failure, CPLE_AppDefined, "Cannot find table %s", m_pszTableName );
+
+        SQLResultFree(&oResultTable);
+        return OGRERR_FAILURE;
+    }
+
+    std::set<std::string> uniqueFields;
+    // Match identifiers with " or ` or no delimiter (and no spaces).
+    static const std::regex sFieldIdentifierRe { R"raw(\s*(["`]([^"`]+)["`])|(([^\s]+)\s).*)raw" };
```

Comment:
I don't think there's a guarantee that the CREATE TABLE will have column identifiers quoted. GDAL will generate such files. But files from other sources could have unquoted identifiers. Not necessarily critical to address this, but perhaps add a comment

Reply:
my idea was to support the three possibilities: unquoted, double quoted and backticks, I should have added test cases for all three.

Reply:
Where comes the backticks alternative? Just curious, I have never seen such yet.

Reply:
I see, SQLite supports unfortunately backticks because MySQL supports them. How about brackets then? 
https://stackoverflow.com/questions/25141090/sqlite-use-backticks-or-double-quotes-with-python.